android/anc/src/main/java/org/smartregister/fhircore/anc/ui/madx/details/NonAncDetailsActivity.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.anc.ui.madx.details
+
+import android.content.Intent
+import android.os.Bundle
+import android.text.SpannableString
+import android.text.style.ForegroundColorSpan
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AlertDialog
+import androidx.core.os.bundleOf
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProvider
+import com.google.android.fhir.FhirEngine
+import com.google.android.material.tabs.TabLayoutMediator
+import org.smartregister.fhircore.anc.AncApplication
+import org.smartregister.fhircore.anc.R
+import org.smartregister.fhircore.anc.data.madx.NonAncPatientRepository
+import org.smartregister.fhircore.anc.data.madx.model.AncPatientDetailItem
+import org.smartregister.fhircore.anc.databinding.ActivityNonAncDetailsBinding
+import org.smartregister.fhircore.anc.ui.madx.details.adapter.ViewPagerAdapter
+import org.smartregister.fhircore.anc.ui.madx.details.form.NonAncDetailsFormConfig
+import org.smartregister.fhircore.engine.ui.base.BaseMultiLanguageActivity
+import org.smartregister.fhircore.engine.ui.questionnaire.QuestionnaireActivity
+import org.smartregister.fhircore.engine.util.extension.createFactory
+
+class NonAncDetailsActivity : BaseMultiLanguageActivity() {
```

Comment:
Same point above about making this generic to PatientDetailsActivity and moving it to engine for reuse